src/Simplify.cpp

```diff
@@ -525,7 +538,11 @@ class Simplify : public IRMutator {
         const Select *select_a = a.as<Select>();
         const Select *select_b = b.as<Select>();
 
-        if (const_int(a, &ia) &&
+        if (is_error_call(call_a)) {
```

Comment:
I'm remembering now why this doesn't break my code, the signed integer overflow was only in bounds inference. So at least the old behavior makes sense.